### ISSUE TITLE
annocheck's run-path check doesn't get skipped with --skip-run-path

### ISSUE THREAD

**Issue by debarshiray** [rpminspect/rpminspect#1296]
I am trying to suppress some of the false positives causing Fedora CI test failures seen in [toolbox](https://src.fedoraproject.org/rpms/toolbox/) Bodhi updates, by looking at these two recent updates:
* `toolbox-0.0.99.4-4.fc38`: https://bodhi.fedoraproject.org/updates/FEDORA-2023-df8fdfca60
* `toolbox-0.0.99.4-5.fc39`: https://bodhi.fedoraproject.org/updates/FEDORA-2023-c5a84ca993

One false positive is that the `DT_RPATH` or `DT_RUNPATH` value of `/run/host%{_libdir}` that's present in `%{_bindir}/toolbox` is [intentional](https://github.com/containers/toolbox/commit/6063eb27b9893994).

I added a [rpminspect.yaml](https://src.fedoraproject.org/rpms/toolbox/blob/rawhide/f/rpminspect.yaml) to address that by allowing `/run/host/usr/lib64` in the standalone runpath test, and by disabling annocheck's own run-path test with `--skip-run-path`.  It worked for the former, but the `--skip-run-path` doesn't seem to be having any effect on the [annocheck](https://artifacts.dev.testing-farm.io/bfc63c70-c9e1-4bf9-a715-72d5b2828316/work-rpminspectg5uu56uf/rpminspect/execute/data/guest/default-0/rpminspect-1/data/viewer.html) test.

I checked that I didn't goof up the hyphens and dashes (eg., https://lwn.net/Articles/947941/) or the annocheck [flags](https://sourceware.org/annobin/annobin.html/Test-run-path.html).

Note that a similar [rpminspect.yaml](https://gitlab.com/redhat/centos-stream/rpms/toolbox/-/blob/c9s/rpminspect.yaml) is also used in CentOS Stream 9.  Maybe I should pay closer attention when I doing my next CentOS Stream 9 build?  :)

**Comment by dcantrell** [comment]
The annocheck block in your rpminspect.yaml is almost there.  I apologize for the format of this configuration block in rpminspect.yaml.  It evolved as annocheck evolved and because rpminspect invokes the command, I have to collect command line options.  There is libannocheck now and rpminspect can link with that, but I still have users put command line options for annocheck in the config file and I just translate those at run time to the library if rpminspect was built that way.

TL;DR -- it has resulted in a slightly confusing annocheck block in the config file.

The definitive resource for what can go in rpminspect.yaml is in the generic template in the source tree:
https://github.com/rpminspect/rpminspect/blob/main/data/generic.yaml

I keep that up to date and commented to explain the blocks.

What you need to do in the package's rpminspect.yaml is this:

```
---
annocheck:
    extra_opts:
        # skip DT_RPATH/DT_RUNPATH checking because toolbox has
        # /run/host in the path
        #
        # skip the stack protection test as well because toolbox
        # contains a region of code compiled without stack protection
        # (crosscall2)
        hardened: --skip-run-path --skip-stack-prot

runpath:
    allowed_paths:
        - /run/host/usr/lib64
```

Change comments as appropriate, but I recommend to package maintainers that you put comments in rpminspect.yaml so that people in the future will know why this package needs specific rpminspect control.

The difference here is that you need to use the **extra_opts** section of the **annocheck** configuration block in order to add additional command line options to the _hardened_ inspection.  I know it's confusing, but this config file format is already out in the wild so I'm just going to live with it.  Hopefully this explanation clarifies things.  Do refer to the [generic.yaml](https://github.com/rpminspect/rpminspect/blob/main/data/generic.yaml) example as a template that you can copy from for future packages.

**Comment by debarshiray** [comment]
Ping, @dcantrell

**Comment by debarshiray** [comment]
Ping, @dcantrell

**Comment by debarshiray** [comment]
Thanks.  I think I managed to [silence](https://bodhi.fedoraproject.org/updates/FEDORA-2024-5d9fd70d84) the run-path check.
